packages/ui/src/components/TextInputV2/index.tsx

```diff
@@ -369,7 +371,7 @@ export const TextInputV2 = forwardRef<HTMLInputElement, TextInputProps>(
             </StyledInputWrapper>
           </Tooltip>
         </div>
-        {error || success || typeof helper === 'string' ? (
+        {error || typeof success === 'string' || typeof helper === 'string' ? (
```

Comment:
Why not use a prop state = {variant: "", text: ""} ? Like this we can handle success, warning, error ? :thinking:

Reply:
Here it can lead to weird behaviour if you pass error and success

Reply:
It won't be consistent this way as depending on which one you choose the text might be facultative or mandatory, also there will be no use case where the error and success will be passed. But let's imagine you do it we defined a specific order: loading > error > success > neutral